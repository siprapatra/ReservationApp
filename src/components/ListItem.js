import React from "react";
import { Text, TouchableOpacity, StyleSheet } from "react-native";
import PropTypes from "prop-types";

const ListItem = ({ item, onItemClick }) => (
  <TouchableOpacity
    style={styles.itemContainer}
    onPress={() => onItemClick(item)}
  >
    <Text style={styles.itemText}>Name: {item.name}</Text>
    <Text style={styles.itemText}>Hotel Name: {item.hotelName}</Text>
    <Text style={styles.itemText}>Departure Date: {item.departureDate}</Text>
    <Text style={styles.itemText}>Arrival Date: {item.arrivalDate}</Text>
  </TouchableOpacity>
);

ListItem.propTypes = {
  item: PropTypes.shape({}).isRequired,
  onItemClick: PropTypes.func
};

ListItem.defaultProps = {
  onItemClick: () => {}
};

const styles = StyleSheet.create({
  itemContainer: {
    marginVertical: 5,
    marginHorizontal: 8,
    borderColor: "black",
    borderWidth: 1,
    padding: 5
  },
  itemText: {
    fontSize: 16,
    color: "black"
  }
});

export default ListItem;
