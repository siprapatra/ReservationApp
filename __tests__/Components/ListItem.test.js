import React from "react";
import renderer from "react-test-renderer";
import ListItem from "../../src/components/ListItem";

var item = {
  name: "Test",
  hotelName: "Test Hotel",
  departureDate: "20 Mar 2019",
  arrivalDate: "22 Mar 2019"
};
describe("Test for ListItem", () => {
  test("should render the Listitem", () => {
    const snap = renderer
      .create(
        <ListItem
          item={item}
          onItemClick={() => {
            console.log("item clicked");
          }}
        />
      )
      .toJSON();
    expect(snap).toMatchSnapshot();
  });
});
