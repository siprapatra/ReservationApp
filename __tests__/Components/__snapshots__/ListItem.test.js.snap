// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for ListItem should render the Listitem 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderColor": "black",
      "borderWidth": 1,
      "marginHorizontal": 8,
      "marginVertical": 5,
      "opacity": 1,
      "padding": 5,
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 16,
      }
    }
  >
    Name: 
    Test
  </Text>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 16,
      }
    }
  >
    Hotel Name: 
    Test Hotel
  </Text>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 16,
      }
    }
  >
    Departure Date: 
    20 Mar 2019
  </Text>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 16,
      }
    }
  >
    Arrival Date: 
    22 Mar 2019
  </Text>
</View>
`;
